module/canvas/detection-modes/detect-evil-and-good.mjs

```diff
@@ -0,0 +1,36 @@
+/**
+ * The detection mode for Detect Evil and Good.
+ */
+export class DetectionModeDetectEvilAndGood extends DetectionMode {
+  constructor() {
+    super({
+      id: "detectEvilAndGood",
+      label: "DND5E.DetectEvilAndGood",
+      type: DetectionMode.DETECTION_TYPES.OTHER,
+      walls: false
+    });
+  }
+
+  /** @override */
+  static getDetectionFilter() {
+    return this._detectionFilter ??= GlowOverlayFilter.create({
+      glowColor: [1, 1, 0, 1]
+    });
+  }
+
+  /** @override */
+  _canDetect(visionSource, target) {
+    if (!(target instanceof Token)) return false;
+    const actor = target.actor;
+    if (!(actor && actor.type === "npc")) return false;
+    const type = actor.system.details.type.value;
+    return type === "aberration"
+      || type === "celestial"
+      || type === "elemental"
+      || type === "fey"
+      || type === "fiend"
+      || type === "undead";
```

Comment:
Maybe a `CONFIG.DND5E` object with the 'good' types and 'evil' types would be nice to have!

Reply:
Yes, we should use a `CONFIG` to store these values rather than hard-coding it in the detection mode.